### DISCUSSION TITLE
AND or OR operator in check?

### DISCUSSION THREAD

**Discussion by bobhenl** [Zrips/CMI#5377] [Enhancement]
Hi, what about adding ANd or OR operator into a check variable in specialized commands? I have this cmd:
```
  role add:
    Cmds:
    - check:$2==null! asConsole! cmi ctext prikazy [playerName] 57
    - check:$2!=null! asConsole! check:$2==Eventer! perm:group.eventer! lp user $1 parent add $2
    - check:$2!=null! asConsole! check:$2==Helper! perm:group.helperm! lp user $1
      parent add $2
    - check:$2!=null! asConsole! check:$2==Helper! perm:group.mod+! lp user $1 parent
      add $2
    - check:$2!=null! asConsole! check:$2==Tr.helper! perm:group.helperm! lp user
      $1 parent add $2
    - check:$2!=null! asConsole! check:$2==Tr.helper! perm:group.mod+! lp user $1
      parent add $2
    - check:$2!=null! asConsole! check:$2==Mod! perm:group.admin! lp user $1 parent
      add $2
    - check:$2!=null! asConsole! check:$2==Streamer! perm:group.mod+! lp user $1 parent
      add $2
    - check:$2!=null! asConsole! check:$2==Youtuber! perm:group.mod+! lp user $1 parent
      add $2
    - check:$2!=null! asConsole! check:$2==Tiktoker! perm:group.mod+! lp user $1 parent
      add $2
    - check:$2!=null! asConsole! check:$2==Instagramer! perm:group.mod+! lp user $1
      parent add $2
    - check:$2!=Eventer! check:$2!=Helper! check:$2!=Tr.helper! check:$2!=Mod! check:$2!=Streamer!
      check:$2!=Youtuber! check:$2!=Tiktoker! check:$2!=Instagramer! asConsole! cmi
      ctext prikazy [playerName] 57
    - check:$2==Helper! check:$2==Eventer! asConsole! cmi msg [playerName] !&c&lRole &f» &eHráči &f$1 &ejsi přidal roli &f$2&e.
    - check:$2!=null! asConsole! cmi msg $1 !&c&lRole &f» &eHráč &f[playerName] &eti přidal roli &f$2&e.
```

But I don't know how to fix the latest two lines to execute only IF $2 is YouTuber, tiktoker, helper, eventer and so on. ->
`    - check:$2==Helper! check:$2==Eventer! asConsole! cmi msg [playerName] !&c&lRole &f» &eHráči &f$1 &ejsi přidal roli &f$2&e.` Would be nice if I could add there AND or OR or something like this.

Or maybe way if the whole output is false, then to cancel continuing somehow?     
```
- check:$2!=Eventer! check:$2!=Helper! check:$2!=Tr.helper! check:$2!=Mod! check:$2!=Streamer!
      check:$2!=Youtuber! check:$2!=Tiktoker! check:$2!=Instagramer! asConsole! cmi
      ctext prikazy [playerName] 57
```
but the problem is that I cannot add # after the one, cause then it won't check for other values (so if I add # to Eventer for ex., the next roles like helper etc. won't be checked....).

Any solution for this? :/

Just a note: Command works as expected, the only problem is that I don't wanna display the message that the player gave & got role when they input there the wrong role that isn't one of the listed. So for ex. when player types /role add <name> sssss, then I don't wanna send that player got role ssss when he didn't get it....

**Comment by CloudeLecaw** [comment]
I also have many cases where i wished i could creat AND or OR condtions chains
Like If its Name1 OR Name2 OR Name3
or When Player chas Permission1 AND Permission2
there combinations would be very endless
normaly these are done by | = OR      + = AND
currently its not possible to combine them like may known from Regex

currently u can have endless equal condition but just one unequal condition in one custom alias

you could do something like this

- check:$2==Eventer! asConsole! cmi ctext prikazy [playerName] 51
- check:$2==Helper! asConsole! cmi ctext prikazy [playerName] 52
- check:$2==Tr.helper! asConsole! cmi ctext prikazy [playerName] 53
- check:$2==Mod! asConsole! cmi ctext prikazy [playerName] 54
- check:$2==Streamer! asConsole! cmi ctext prikazy [playerName] 55
- check:$2==Youtuber! asConsole! cmi ctext prikazy [playerName] 56
- check:$2==Tiktoker! asConsole! cmi ctext prikazy [playerName] 57
- check:$2==Instagramer! asConsole! cmi ctext prikazy [playerName] 58
- check:$2!=Instagramer! asConsole! cmi ctext prikazy [playerName] 1
- 
All get Page 1 in that case except Instagrmer, also some get page 51 or 52 53 54 55 56 57 58 if they meet the condition.

for =! you just can doone logical condiion at the moment
sepeating by | (OR)  or  + (AND) like in Regex would be a nice thing i agree with this

Then it would look like this


- check:$2!=Eventer|Helper|Tr.helper|Mod|Streamer|Youtuber|Tiktoker|Instagramer! asConsole! cmi ctext prikazy [playerName] 57

or maybe

- check:$2!=Eventer|!=Helper|!=Tr.helper!=|Mod|!=Streamer|!=Youtuber|!=Tiktoker|!=Instagramer! asConsole! cmi ctext prikazy [playerName] 57

but for not its not possible to have more then one != condition at once becaus this one matches aggainst all other at the moment

**Comment by CloudeLecaw** [comment]
Abount your example is cant work because your checking two times for $2 in one line and this is not possible
Example:
`check:$2!=null! asConsole! check:$2==Eventer!`
$2 cant be NULL (norhting) and in same place the value Eventer
May you should correct your example first..

NULL != Eventer

Also finish all Checks and Condtions before using asConsole or asPlayer
if u used asConsole one then other condition after are a bit.... useless like permission
console has all permissions....
So what you tried here?
Something like this?
`- check:$1!=null! check:$2==Eventer! perm:group.eventer! asConsole! lp user $1 parent add $2`
Its recommand to check for the permission first before looking into arg one or two
`- perm:group.eventer! check:$1!=null! check:$2==Eventer! asConsole! lp user $1 parent add $2`
This would make more sense
It sais $1 cant be no name and $2 has to be Eventer to run the command behind
then it executes` lp user NAME parent add Eventer` as console if the executer has permisison `perm:group.eventer`

Here what you can do; (i remove the != chain part because its not possible at the moment that way and i renamed your mod+ + and | are not recomment to use)

```
 role add:
    Cmds:
    - check:$1==null! asConsole! cmi ctext prikazy [playerName] 57
    - check:$2!=null! asConsole! cmi msg $1 !&c&lRole &f» &eHráč &f[playerName] &eti přidal roli &f$2&e.
    - check:$2==Eventer! asConsole! cmi msg [playerName] !&c&lRole &f» &eHráči &f$1 &ejsi přidal roli &f$2&e.
    - check:$2==Helper! asConsole! cmi msg [playerName] !&c&lRole &f» &eHráči &f$1 &ejsi přidal roli &f$2&e.
    - perm:group.admin! check:$1!=null! check:$2==Mod! asConsole! lp user $1 parent add $2
    - perm:group.eventer! check:$1!=null! check:$2==Eventer! asConsole! lp user $1 parent add $2
    - perm:group.helperm! check:$1!=null! check:$2==Helper! asConsole! lp user $1 parent add $2
    - perm:group.helperm! check:$1!=null! check:$2==Tr.helper! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Helper! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Instagramer! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Streamer! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Tiktoker! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Tr.helper! asConsole! lp user $1 parent add $2
    - perm:group.mod! check:$1!=null! check:$2==Youtuber! asConsole! lp user $1 paren add $2
```
